matchzoo/contrib/models/esim.py

```diff
@@ -0,0 +1,234 @@
+        mlp_input = keras.layers.Dropout(
+            rate=self._params['dropout_rate'])(mlp_input)
+        # Output layer
+        mlp = self._make_multi_layer_perceptron_layer()(mlp_input)
+        mlp = keras.layers.Dropout(
+            rate=self._params['dropout_rate'])(mlp)
+        inputs = [input_left, input_right]
+        # mlp = keras.layers.Lambda(lambda x: K.tf.Print(x, [x]))(mlp)
+        x_out = self._make_output_layer()(mlp)
+        self._backend = keras.Model(inputs=inputs, outputs=x_out)
+
+    def _attention(self, encoded_left,
+                   left_mask,
+                   encoded_right,
+                   right_mask):
+        # Interaction
+        embed_cross = keras.layers.dot([encoded_left, encoded_right], axes=2)
+        cross_shape = [self._params['input_shapes'][0][0], self._params['input_shapes'][1][0]]
+        embed_cross = keras.layers.Reshape(cross_shape)(embed_cross)
+        left_to_right_attn = self._masked_softmax(embed_cross, right_mask,
+                                                  self._params['input_shapes'][0][0],
+                                                  self._params['input_shapes'][1][0])
+
+        right_to_left_attn = self._masked_softmax(keras.layers.Permute([2, 1])(embed_cross), left_mask,
+                                                  self._params['input_shapes'][1][0],
+                                                  self._params['input_shapes'][0][0])
+
+        attended_left = self._weighted_sum(left_to_right_attn,
+                                           encoded_right,
+                                           left_mask)
+        attended_right = self._weighted_sum(right_to_left_attn,
+                                            encoded_left,
+                                            right_mask)
+        return attended_left, attended_right
+
+    # def _masked_softmax(self, input, mask, att_len, base_len, epsilon=1e-8):
+    #
+    #     def tile_mask(x):
+    #         res = K.tile(K.expand_dims(x, axis=1), [1, att_len, 1])
+    #         return res
+    #
+    #     tiled_mask = keras.layers.Lambda(tile_mask)(mask)
+    #     reshaped_mask = keras.layers.Lambda(lambda x: K.reshape(x, [-1, base_len]))(tiled_mask)
+    #     reshaped_input = keras.layers.Lambda(lambda x: K.reshape(x, [-1, base_len]))(input)
+    #     softmax_res = keras.layers.Softmax() \
+    #         (keras.layers.multiply([reshaped_input, reshaped_mask]))
+    #     softmax_res = keras.layers.multiply([softmax_res, reshaped_mask])
+    #     softmax_res = keras.layers.Lambda(lambda x: x / K.sum(x, axis=-1, keepdims=True) + epsilon)(softmax_res)
+    #     output = keras.layers.Lambda(lambda x: K.reshape(x, [-1, att_len, base_len]))(softmax_res)
+    #     output = keras.layers.Lambda(lambda x: K.tf.Print(x, [x]))(output)
+    #     return output
+
+    def _masked_softmax(self, input, mask, att_len, base_len):
+        inf = 1e8

```

Comment:
1e8 and 1e7 define as the same parameter on the top of the model file.
such as `INF = 1e7`